stock_storage_type/models/stock_location.py

```diff
@@ -153,125 +233,98 @@ def get_storage_locations(self, products=None):
-        else:
-            return location_storage_type
-
-    def _filter_capacity(self, location_storage_type, quants):
-        if self._max_height_allowed(
-            location_storage_type, quants
-        ) and self._max_weight_allowed(location_storage_type, quants):
-            return location_storage_type
-        else:
-            return self.env["stock.location.storage.type"]
-
-    def _max_height_allowed(self, location_storage_type, quants):
-        height = quants.package_id.height
-        max_height = location_storage_type.max_height
-        return not (max_height and height and height > max_height)
-
-    def _max_weight_allowed(self, location_storage_type, quants):
-        pack_weight = quants.package_id.pack_weight
-        max_weight = location_storage_type.max_weight
-        return not (max_weight and pack_weight and pack_weight > max_weight)
-
-    def _existing_quants(self, products=None, lot=None):
-        base_domain = [("location_id", "child_of", self.id)]
-        domain = self._prepare_existing_domain(base_domain, products=products, lot=lot)
-        return self.env["stock.quant"].search(domain, limit=1)
-
-    def _existing_planned_move_lines(self, products=None, lot=None):
-        base_domain = [
-            ("location_dest_id", "child_of", self.id),
-            ("state", "in", ("waiting", "partially_available", "assigned")),
+        pertinent_loc_storagetype_domain = [
+            ('id', 'in', compatible_location_storage_types.ids),
+            ('package_storage_type_ids', '=', package_storage_type.id),
         ]
-        domain = self._prepare_existing_domain(base_domain, products=products, lot=lot)
-        return self.env["stock.move.line"].search(domain, limit=1)
+        if quants.package_id.height:
+            pertinent_loc_storagetype_domain += [
+                '|',
+                ('max_height', '=', 0),
+                ('max_height', '>', quants.package_id.height),
+            ]
+        if quants.package_id.pack_weight:
+            pertinent_loc_storagetype_domain += [
+                '|',
+                ('max_weight', '=', 0),
+                ('max_weight', '>', quants.package_id.pack_weight),
+            ]
 
-    def _existing_planned_moves(self, products=None):
-        if self.child_ids:
-            # If a location is a leaf, it's a "bin", we know that the move line will
-            # be in this exact location. If it has sub-locations, we can't be sure
-            # where it will go and we don't want to restrict based on this.
-            return self.env["stock.move"].browse()
-        base_domain = [
-            ("location_dest_id", "=", self.id),
-            ("state", "in", ("waiting", "confirmed")),
-        ]
-        domain = self._prepare_existing_domain(base_domain, products=products)
-        return self.env["stock.move"].search(domain, limit=1)
+        pertinent_loc_storage_types = StockLocationStorageType.search(
+            pertinent_loc_storagetype_domain
```

Comment:
The building of `pertinent_loc_storagetype_domain` should be extracted in a method to allow customizations.

Reply:
done